pkg/process/checks/checks.go

```diff
@@ -34,31 +34,49 @@ type SysProbeConfig struct {
 // processed in another way (e.g. printed for debugging).
 // Before checks are used you must called Init.
 type Check interface {
-	Init(syscfg *SysProbeConfig, info *HostInfo) error
 	Name() string
-	RealTime() bool
-	Run(groupID int32) ([]model.MessageBody, error)
+	IsEnabled() bool
+	Realtime() bool
+	Init(syscfg *SysProbeConfig, info *HostInfo) error
+	SupportsRunOptions() bool
```

Comment:
Do you mind adding a small comment about what `SupportsRunOptions` is used for? My worry is that the next person might not understand it the next time we want to add a check

Reply:
Thank you, yes! 👍 This was something that I wanted to do - add a comment for each method here.